src/common.js

```diff
@@ -57,18 +58,28 @@ exports.error = error;
 //@
 //@ Examples:
 //@
-//@ ```
+//@ ```javascript
 //@ var foo = ShellString('hello world');
 //@ ```
 //@
 //@ Turns a regular string into a string-like object similar to what each
 //@ command returns. This has special methods, like `.to()` and `.toEnd()`
-var ShellString = function (str, stderr) {
-  var that = new String(str);
-  that.to = wrap('to', _to, {idx: 1});
-  that.toEnd = wrap('toEnd', _toEnd, {idx: 1});
-  that.stdout = str;
+var ShellString = function (stdout, stderr) {
+  var that;
+  if (stdout instanceof Array) {
+    that = stdout;
+    that.stdout = stdout.join('\n')+'\n';
+  } else {
+    that = new String(stdout);
+    that.stdout = stdout;
+  }
   that.stderr = stderr;
+  that.to    = function() {wrap('to', _to, {idx: 1}).apply(that.stdout, arguments); return that;};
+  that.toEnd = function() {wrap('toEnd', _toEnd, {idx: 1}).apply(that.stdout, arguments); return that;};
+  that.cat   = function() {return shell.cat.apply(that.stdout, arguments);};
```

Comment:
I tried binding instead, but it wound up being even more complicated, and still wasn't working completely correctly. I think it's simplest to just leave as closures.